### ISSUE TITLE
Opening a new buffer in Glowing Bear switches weechat to it, keeping messages marked read

### ISSUE THREAD

**Issue by SimonSapin** [glowing-bear/glowing-bear#940]
Steps to reproduce:

* In Glowing Bear, join a new channel, or start a new private conversation such as a new buffer is opened in weechat
* Switch Glowing Bear away from this channel/conversation
* Wait until new messages come in, in that channel/conversation

Expected result: the unread count shows the number of messages since switching away.

Actual result: the unread count is occasionally (?) reset to zero because weechat is "showing" that buffer, and so messages are considered "read". Important messages may be missed.

Work around: SSH into the server, attach the GNU screen session, and run `/buffer 1` to switch weechat back to a buffer when keeping track of unread messages is unimportant. Don’t forget every time a new buffer is created.

Possible fix (?): creating a new buffer from Glowing Bear should not make weechat switch to that buffer.

**Comment by lorenzhs** [comment]
What's your WeeChat version? This shouldn't happen with recent WeeChat.

**Comment by lorenzhs** [comment]
See https://github.com/glowing-bear/glowing-bear/blob/eaffb17c5a24454d90df384915a692d87539e6eb/js/glowingbear.js#L487

It shouldn't switch channels at all, but requires WeeChat 1.2 or later to prevent the switching for queries

**Comment by lorenzhs** [comment]
Of course if you manually `/join` or `/query`, we don't rewrite your commands. Use `/join -noswitch` to prevent WeeChat from switching.

**Comment by lorenzhs** [comment]
Technically it would be possible to rewrite commands in https://github.com/glowing-bear/glowing-bear/blob/92d707a51d92adcfc099e36ece8b711e1454f97f/js/inputbar.js#L121 but doing so silently is something I would be *very* uncomfortable with. Maybe we could add a prompt the first time asking the user whether we're allowed to rewrite join commands?

**Comment by SimonSapin** [comment]
I use weechat version 1.0.1-1+deb8u1 from Debian. I see that 1.6-1+deb9u1~bpo8+1 exists in backports, I’ll try that. Also I just learned about `-noswitch`, thanks.

I can start a query by clicking on a nickname in the list. I assume this goes through the code you first linked. For joining a channel however I don’t know a way other than typing `/join #foo`. What’s the other one? (I assume there is at least one, since that code is written for it.)

**Comment by lorenzhs** [comment]
If someone mentions a channel in a message, that's turned into a link that opens that channel when you click it. This mechanism uses `-noswitch`.

**Comment by SimonSapin** [comment]
I see, thanks. Joining a new channel is not necessarily when someone mentions it, so I think using `/join` explicitly is still necessary sometimes. On the other hand, I can’t think of a case where switching weechat when creating a new buffer from Glowing Bear is ever desirable. So, while I understand the reluctance to implicitly rewrite commands, I think it should be done in this case. This problem is subtle and there is no easy way to figure out what’s going on, or that `-noswitch` exists. Re-purposing this issue.

**Comment by lorenzhs** [comment]
Yes obviously I join most channels with `/join` too, but when clicking a channel link in GB the user expects GB to do the right thing, so we send `/join -noswitch #foobar`. It's a different beast to tinker with commands the user explicitly entered, which is why we aren't currently doing that. @torhve @cormier thoughts on asking the user whether we can rewrite join/query commands to add `-noswitch` the first time they use such a command?

**Comment by Techman-** [comment]
You could make it a setting. I'm not sure exactly what wording it should be, though. CC: @lorenzhs @torhve

**Comment by lorenzhs** [comment]
We've been over this many times, no settings for things that shouldn't need one. Maybe a non-modal warning could be shown if `/j` / `/join` / `/query` without `-noswitch` is detected, with a "click here to add `-noswitch`"?
